rustual-boy-middleware/src/color.rs

```diff
@@ -0,0 +1,69 @@
+//! Some color utilities that are useful for implementing the anaglyph modes
+
+/// Represents a color
+pub struct Color {
+    r: f32,
+    g: f32,
+    b: f32,
+}
+
+impl From<(f32, f32, f32)> for Color {
+    /// Convert a tuple of RGB in the [0, 1] range to a color
+    fn from((r, g, b): (f32, f32, f32)) -> Color {
+        Color {r: r, g: g, b: b}
```

Comment:
nit: Spaces are preferred between the braces and components: `{ r: r, g: g, b: b }` but since we don't usually run rustfmt (since I haven't configured it correctly for some cases) these are hard to enforce by hand.